src/js_native_api_v8.cc

```diff
@@ -2452,8 +2452,16 @@ napi_status NAPI_CDECL napi_check_object_type_tag(napi_env env,
     napi_type_tag tag;
     val.As<v8::BigInt>()->ToWordsArray(
         &sign, &size, reinterpret_cast<uint64_t*>(&tag));
-    if (size == 2 && sign == 0)
-      *result = (tag.lower == type_tag->lower && tag.upper == type_tag->upper);
+    if (sign == 0) {
+      if (size == 2) {
+        *result =
+            (tag.lower == type_tag->lower && tag.upper == type_tag->upper);
+      } else if (size == 1) {
+        *result = (tag.lower == type_tag->lower);
```

Comment:
IIUC, `type_tag->upper` should also be checked when the size is 1.

```suggestion
        *result = (tag.lower == type_tag->lower && 0 == type_tag->upper);
```

And would you mind adding a test verifying that tagging an object with the upper value as 0 and checking the type tag with the upper value as a non-zero value?

Reply:
Thanks for the review. I also added more tests. I guess that the test you mentioned is the following case, PTAL.

```js
assert.strictEqual(test_object.CheckTypeTag(4, obj3), false);
```